### PR TITLE
:book: Content organization

### LINKED ISSUE #1305: Collect all 1.0.0 draft docs into a presentable hierarchy

### PR DESCRIPTION
**Pull request by perdasilva**
# Description

Closes #1305

This PR takes a first stab at organizing the content in docs/drafts into some kind of coherent structure.
There's still much to do in terms of making the documents more uniform and prettying them up.

## Reviewer Checklist

- [ ] API Go Documentation
- [ ] Tests: Unit Tests (and E2E Tests, if appropriate)
- [ ] Comprehensive Commit Messages
- [ ] Links to related GitHub Issue(s)

### CODE REVIEW

File: docs/index.md [hunk 1]
```diff
@@ -20,30 +27,18 @@ The first iteration of OLM, termed OLM v0, included several concepts and feature
 Since its initial release, OLM has helped catalyse the growth of Operators throughout the Kubernetes ecosystem. [OperatorHub.io](https://operatorhub.io/)
 is a popular destination for discovering Operators, and boasts over 300 packages from many different vendors.
 
-# Why are we building OLM v1?
+## Why are we building OLM v1?
 
 OLM v0 has been in production for over 5 years, and the community to leverage this experience and question the initial
```
Comment (grokspawn) [inline docs/index.md:32]: #EditorialNotOrganizational

`community to leverage`?

I'm not sure what the original is attempting to express with this sentence.  Maybe 
>... for over 5 years, and this has provided the community with ample time to leverage this experience and question the initial goals ...

?

File: docs/project/olmv1_community.md [hunk 2]
```diff
@@ -0,0 +1,11 @@
+In this next iteration of OLM, the community has also taken care to make it as contributor-friendly as possible, and welcomes new contributors.
+The project is tracked in a [GitHub project](https://github.com/orgs/operator-framework/projects/8/),
```
Comment (grokspawn) [inline docs/project/olmv1_community.md:2]: Do we need to note that this is a CNCF project or is that self-evident?

File: docs/project/olmv1_architecture.md
Comment (grokspawn) [inline docs/project/olmv1_architecture.md:1]: It appears that the bulleted lists are not getting rendered properly in the [netlify preview](https://github.com/operator-framework/operator-controller/pull/1324/files#diff-171794fb2ee097c9ecf4da40b621754333a472fd9befe64047960ff202f535d9R61-R69).  I can't tell if this is an actual issue or a netlify-only problem.  

We need to check these once this merges, and IFF it is broken, then we need an issue to fix.

**Review by grokspawn — COMMENTED** [review]
Only nits.  I have some concerns about the evident need to hide ToC (since I don't know why), but it doesn't rise to the level of challenging the move. 

Separately, we need some love on the contents.  In general I've tried to flag comments about existing content in a clear way so we can keep this PR focused on the portal organization / format instead. I've identified one existing editorial issue I think we can easily solve though, and a potential netlify problem when previewing bulleted lists that I ran into with semver template docs in the past.

File: docs/project/olmv1_architecture.md
Comment (perdasilva) [reply docs/project/olmv1_architecture.md:1]: Yeah, I think it reacts poorly to not having an empty line before the start of the list =S

File: docs/index.md (hunk 1 above)
Comment (perdasilva) [reply docs/index.md]: Probably my fault. Sometimes I keep iterating over a paragraph to the point where it doesn't make sense anymore XD I tried fixing it up

File: docs/project/olmv1_community.md (hunk 2 above)
Comment (perdasilva) [reply docs/project/olmv1_community.md]: Added ^^

File: docs/project/olmv1_architecture.md
Comment (perdasilva) [reply docs/project/olmv1_architecture.md:1]: I've tried to go through each of the documents and ensure there's an empty line on top of lists. I think I Ash Ketchup'ed them

**Review by LalatenduMohanty — APPROVED** [review]
/lgtm

**Review by grokspawn — APPROVED** [review]

**Pull request merged**